tests/unit/tf/data_augmentation/test_negative_sampling.py

```diff
@@ -81,15 +82,28 @@ def test_calling(self, music_streaming_data: Dataset, to_dense: bool, tf_random_
         )
 
     def test_in_model(self, music_streaming_data: Dataset, tf_random_seed: int):
+        dataset = music_streaming_data
+        schema = dataset.schema
+
+        @tf.keras.utils.register_keras_serializable(package="merlin.models")
         class Training(tf.keras.layers.Layer):
             def call(self, inputs, training=False):
                 return training
 
+        @tf.keras.utils.register_keras_serializable(package="merlin.models")
+        class Identity(tf.keras.layers.Layer):
```

Comment:
@marcromeyn The Cond layer throws an exception without this, I think because the return value is different depending on whether or not targets is None or not.

Reply:
Can’t we replace the `NoOp` inside the `Cond` with this? So that this is used by default and our users don’t have it call it explicitly?

Reply:
I don't think we can because this logic is specific to the way the sampling layer works. sometimes an identity function and sometimes transforms into a tuple